lib/active_merchant/billing/gateways/cyber_source.rb

```diff
@@ -931,6 +933,13 @@ def add_auth_wallet(xml, payment_method, options)
             xml.tag!('xid', Base64.encode64(cryptogram[20...40])) if cryptogram.bytes.count > 20
             xml.tag!('reconciliationID', options[:reconciliation_id]) if options[:reconciliation_id]
           end
+        when :discover
+          return unless options[:enable_cybs_discover_apple_pay]
+
+          xml.tag! 'ccAuthService', { 'run' => 'true' } do
+            xml.tag!('cavv', payment_method.payment_cryptogram) unless commerce_indicator
+            xml.commerceIndicator commerce_indicator.nil? ? ECI_BRAND_MAPPING[brand] : commerce_indicator
```

Comment:
Do we also want to pass `options[:reconciliation_id]`?

Reply:
No I don't think so. That is noted in the visa docs, but not the discover docs.

Reply:
in the cybersource docs, it seems like in this context DIPB is what is required. https://developer.cybersource.com/content/dam/docs/cybs/en-us/apple-pay/developer/smartpay/so/applepay.pdf
![image](https://github.com/user-attachments/assets/8606dfff-8708-41cd-ae67-47551634b3a3)